docs/8-the-supporting-acts/2-linting.md

```diff
@@ -0,0 +1,170 @@
+    /opt/app-root/src/jukebox-ui/chart/templates/deployment.yaml: (object: <no namespace>/jukebox-ui apps/v1, Kind=Deployment) container "jukebox-ui" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in the container securityContext.)
+
+    Error: found 1 lint errors
+    </code></pre>
+    </div>
+
+    _..which is luckily not a problem for OpenShift users because OpenShift by default mounts the container's root filesystem as read-only, so we don't have to specify this in our deployments 🎉_
+
+    If you would like to extend kube-linter checks of the common Kubernetes objects or extend the checks to Custom Resources such as `InferenceService`, please look at [product documentation](https://docs.kubelinter.io/#/configuring-kubelinter?id=run-custom-checks).
+
+### Helm Linting
+
+We can also use `helm lint` to examine a chart for possible issues. If helm linter encounters things that will cause the chart to fail installation, it will throw error messages. If it encounters issues that break with convention or recommendation, it will let you know with a Warning.
+
+1. Let's try **helm linter** out on model deployment chart folder before adding it as a step to our pipeline. 
+
+    ```bash
+    cd /opt/app-root/src/
+    cd /opt/app-root/src/mlops-helmcharts/charts/model-deployment/
+    helm lint music-transformer-with-feast
+    ```
+
+    Great! No charts failed 👏
+
+    <div class="highlight" style="background: #f7f7f7; overflow-x: auto; padding: 10px;">
+    <pre><code class="language-bash">
+    $ helm lint music-transformer-with-feast 
+    ==> Linting music-transformer-with-feast
+    [INFO] Chart.yaml: icon is recommended
+
+    1 chart(s) linted, 0 chart(s) failed
+    </code></pre>
+    </div>
+
+## Extend Pipeline
+
+
+We’re adding a new Task to our Tekton pipeline to perform linting before triggering the data science pipeline. This ensures that we catch any issues with code or deployment files early on, saving time and resources by avoiding unnecessary training runs when the code doesn’t meet required standards.
+
+1. Open up `mlops-gitops/toolings/ct-pipeline/config.yaml` and add `linting: true` flag to introduce [linting task](https://<GIT_SERVER>/<USER_NAME>/mlops-helmcharts/src/branch/main/charts/pipelines/templates/tasks/linting.yaml).
+
+    ```yaml
+    chart_path: charts/pipelines
+    USER_NAME: <USER_NAME>
+    cluster_domain: <CLUSTER_DOMAIN>
+    git_server: <GIT_SERVER> 
+    alert_trigger: true 
+    linting: true # 👈 add this
+    ```
+
+2. Commit the changes to the repo:
+
+    ```bash
+    git pull
+    cd /opt/app-root/src/mlops-gitops
+    git add .
+    git commit -m "☀️ linting task is added ☀️"
+    git push
+    ```
+
+3. Kick off a pipeline with an empty commit to see the changes on the pipeline:
+
+    ```bash
+    cd /opt/app-root/src/jukebox
+    git commit --allow-empty -m "🩴 trigger pipeline for linting 🩴"
```

Comment:
Won't this cause a crash when trying to create a registry entry?
Or if on purpose, maybe we should call that out? :)

Reply:
empty commit has a new commit ID, shouldn't be a problem 🫣

Reply:
Ah great!